### ISSUE TITLE
[NavigationDrawer] - Mini navigation drawer interacts badly with react-router v4

### ISSUE THREAD

**Issue by stickfigure** [mlaursen/react-md#344]
### Description

The active state isn't following the clicked menu item correctly - it takes two clicks. The first click changes the URL bar and updates the view area, the second click changes the active state (set via `<Route children=`. Here's a video of what it looks like:

http://take.ms/8GvU7

The code for the link looks roughly like this:

```
		<Route path={routePath} children={ ({match}) => (
			<ListItem active={!!match} component={Link} to={destination} {...props} />
		)} />
```

This works perfectly with a fully expanded sidebar - it's only the mini toolbar mode that has an issue. When debugging the mini toolbar mode, it appears that the first click still passes the old `match` values in. I couldn't figure out why.

### Version
- React 15.4.2
- React-MD 1.0.12
- react-router 4.1.1

**Comment by cfilipov** [comment]
This is happening for me as well when I use `NavigationDrawer.DrawerTypes.PERSISTENT`. Problem also goes away when switching to the default full drawer. This should easily reproducible since I'm basically using the [react-router-v4 example code](https://github.com/mlaursen/react-md/tree/master/examples/with-react-router-v4) with the aforementioned `drawerType`.

### Version 
* React 15.5.4
* React-md 1.0.13
* React-router 4.1.1

**Comment by MaryCaroline** [comment]
I'm facing the same. Is there a solution or workaround for this issue?

**Comment by Fefe-Hern** [comment]
I've managed to achieve a sort of workaround by moving `<BrowserRouter>` inside of the NavigationDrawer tags.

**Comment by mlaursen** [comment]
This is a bit late, but I'm going to close this issue now since it should be available since v2.8.0 (I think this is when I added mini support). The only remaining issues for mini navigation items appear to be #1101 and #1082
